applications/luci-app-filemanager/htdocs/luci-static/resources/view/system/filemanager.js

```diff
@@ -0,0 +1,1980 @@
+            if (selectedItems.size > 0) {
+                deleteSelectedButton.style.display = '';
+            } else {
+                deleteSelectedButton.style.display = 'none';
+            }
+        }
+    },
+    // Update the "Select All" checkbox state
+    updateSelectAllCheckbox: function() {
+        // Set the "Select All" checkbox state based on individual checkboxes
+        var selectAllCheckbox = document.getElementById('select-all-checkbox');
+        var allCheckboxes = document.querySelectorAll('.select-checkbox');
+        var allChecked = true;
+        allCheckboxes.forEach(function(checkbox) {
+            if (!checkbox.checked) {
+                allChecked = false;
+            }
+        });
+        if (selectAllCheckbox) {
+            selectAllCheckbox.checked = allChecked;
+        }
+    },
+    // Handler for the "Select All" checkbox change
+    handleSelectAllChange: function(ev) {
+        // Logic to select or deselect all files
+        var self = this;
+        var selectAllCheckbox = ev.target;
+        var allCheckboxes = document.querySelectorAll('.select-checkbox');
+        selectedItems.clear();
+        allCheckboxes.forEach(function(checkbox) {
+            checkbox.checked = selectAllCheckbox.checked;
+            var filePath = checkbox.getAttribute('data-file-path');
+            if (selectAllCheckbox.checked) {
+                selectedItems.add(filePath);
+            }
+        });
+        this.updateDeleteSelectedButton();
+    },
+    // Handler for deleting selected items
+    handleDeleteSelected: function() {
+        // Delete selected files and directories
+        var self = this;
+        if (selectedItems.size === 0) {
+            return;
+        }
+        if (!confirm(_('Are you sure you want to delete the selected files and directories?'))) {
+            return;
+        }
+        var promises = [];
+        selectedItems.forEach(function(filePath) {
+            promises.push(fs.remove(filePath).catch(function(err) {
+                ui.addNotification(null, E('p', _('Failed to delete %s: %s').format(filePath, err.message)), 'error');
+            }));
+        });
+        Promise.all(promises).then(function() {
+            ui.addNotification(null, E('p', _('Selected files and directories deleted successfully.')), 'info');
+            selectedItems.clear();
+            self.updateDeleteSelectedButton();
+            self.loadFileList(currentPath).then(function() {
+                self.initResizableColumns();
+            });
+        }).catch(function(err) {
+            ui.addNotification(null, E('p', _('Failed to delete selected files and directories: %s').format(err.message)), 'error');
+        });
+    },
+    // Function to load the file list
+    loadFileList: function(path) {
+        // Get the list of files and display them in the table
+        var self = this;
+        selectedItems.clear();
+        return getFileList(path).then(function(files) {
```

Comment:
Can you format the columns such that the space between the left window edge and filename isn't zero. A small amount of space is needed.

Reply:
Fixed